### ISSUE TITLE
Incorrect serialization of MapAxis

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#1887]
When writing a `Map` to disk the properties of the `MapAxis` can be incorrectly serialized, in particular `interp`. 

```
from gammapy.maps import MapAxis, WcsGeom, Map
import numpy as np
energy_axis = MapAxis(nodes=np.logspace(-2., 2, 5), unit='TeV', name='energy', node_type='center', interp='lin')
geom = WcsGeom.create(skydir=(0,0), binsz=1, npix=10, axes=[energy_axis])
m = Map.from_geom(geom,unit='m2')
m.write('test.fits',overwrite=True)
m2=Map.read('test.fits')
print(m.geom.axes[0])
print(m2.geom.axes[0])
```
yields:
```
MapAxis

	name       : energy    
	unit       : 'TeV'     
	nbins      : 5         
	node type  : center    
	center min : 1.0e-02 TeV
	center max : 1.0e+02 TeV
	interp     : lin       

MapAxis

	name       : energy    
	unit       : 'TeV'     
	nbins      : 5         
	node type  : center    
	center min : 1.0e-02 TeV
	center max : 1.0e+02 TeV
	interp     : log       
```

Ideas @cdeil, @adonath @AtreyeeS ?

**Comment by cdeil** [comment]
@registerrier - Good catch! Very important that we fix this (and any other) FITS serialisation issues.

For `MapAxis`, I think `interp` is the only property that isn't stored in FITS?

As far as I can see, nothing is said here concerning `interp`:
https://gamma-astro-data-formats.readthedocs.io/en/latest/skymaps/index.html

I guess we should invent a header key for this, add it to the spec, and then to Gammapy.

cc @woodmd

**Comment by cdeil** [comment]
Looking at the FITS and WCS standard, or similar packages like https://specutils.readthedocs.io or https://spectral-cube.readthedocs.io and see how they serialise frequency or energy axes would also be good, before we make a decision how to store axis interpolation mode in gamma-ray astronomy.

If it's not clear from documents if / what solution exists for extra axis interpolation modes, we could also ask on https://lists.nasa.gov/mailman/listinfo/open-gamma-ray-astro or somewhere else for thoughts.

**Comment by registerrier** [comment]
The error appears only for `ENERGY` axes names. I suppose it comes from the special treatment used for compatibility with Fermi maps.
Removing the axis name in the example above yields correct results.

**Comment by AtreyeeS** [comment]
I don't think that the error appears only for `energy` axis names. The `interp` keyword is somewhere hardcoded, and that is what it picks up.
If the axis is named `energy`, it picks up `interp` as `log`.
For any other name (or no name), the `interp` is `lin`.
For example, `print(m2.geom.axes[0])` never shows  ` interp : sqrt`

**Comment by registerrier** [comment]
At the moment, the default behavior implemented in `find_and_read_bands` in [`geom.py`](https://github.com/gammapy/gammapy/blob/ccc0de85a1e5a344b1eddb10fac588480ba070a3/gammapy/maps/geom.py#L83) is to set `interp = "lin"`except if a column is named `ENERGY` or `E_MIN`.

Normally with the columns store centers and edges, so one should be able to guess the type from the columns values. There should not be the necessity to store the `interp` keyword. 

I suppose the issue arises because Fermi data cubes have only some of the columns.

**Comment by AtreyeeS** [comment]
Hello! Working on this, I am a bit confused about how to deal with the time axis. Currently, `Maps` is unable to even read files for `name='time', node_type =edges` axis. This is because there is a confusion between the names `T_MIN` and `TIME_MIN` (Both are partly and inconsistently implemented in gammapy presently.)

Looking at https://gamma-astro-data-formats.readthedocs.io/en/latest/skymaps/index.html?highlight=time , I am a bit uncertain if `T_MIN` and `T_MAX` are special names which **must** be imposed (like `E_MIN` and `E_MAX` in case of `energy`), or is this just an example for GADF? Specifically, for integral `time` axes, should be impose the  name to be `T_MIN`, or let it automatically be `TIME_MIN` ? As far as I understood, only energy axis has to be a special case for backward compatibility with Fermi STs, so we can remove special handling for time.

@registerrier , @cdeil does this sound OK to you?

**Comment by cdeil** [comment]
@AtreyeeS - I think `T_MIN` and `T_MAX` are just given as examples in the spec, and it would be simpler and more consistent to rename to `TIME_MIN` and `TIME_MAX` in https://gamma-astro-data-formats.readthedocs.io/en/latest/skymaps/index.html

On the Gammapy code, I think all that needs to be done is to delete these two lines?

https://github.com/gammapy/gammapy/blob/089d552885256c560c3febdb4610b98b4e708bf0/gammapy/maps/geom.py#L69-L70

> Currently, Maps is unable to even read files for name='time', node_type =edges axis.

Could you please first add a failing test for this (leave a comment pointing to this issue saying that it's a regression test), and then make the code change?

Please send a separate small PR for the time axis change in Gammapy and in the spec. It's easier to review and discuss, and leaves a good changelog.

Overall I think the time axis does need special treatment. There are the MJDREF et al header keys that are needed to specify a reference time (see https://gamma-astro-data-formats.readthedocs.io/en/latest/general/time.html) and that we use everywhere else (events, light curves). We currently don't support or test time axes properly. But that's just another task, and we don't have a need for that at the moment. If time is special-cases in axis I/O, we could do the same as for energy and thus we could use TIME / T_MIN / T_MAX. But overall I think TIME / TIME_MIN / TIME_MAX is just more consistent, it's already what I put here: https://gamma-astro-data-formats.readthedocs.io/en/latest/lightcurves/index.html
So let's do the small change to TIME_MIN / TIME_MAX now, and leave adding proper time axis for later (or never if the use case doesn't arise).

**Comment by cdeil** [comment]
Fixed by @AtreyeeS in #1926 .

I'll make a small edit in the spec to clarify.

**Comment by cdeil** [comment]
I just made this very small edit: 9da2bc58066a0f11760d93c2a52969b696ba54f2

Of course, if someone is willing to send a PR to clarify and extend the spec to explain about interp and node_type, that would be welcome.
